src/librustc_driver/pretty.rs

```diff
@@ -801,6 +807,48 @@ pub fn pretty_print_input(sess: Session,
             })
         }
 
+        (PpmMir, None) => {
+            debug!("pretty printing MIR for whole crate");
+            let ast_map = ast_map.expect("--unpretty mir missing ast_map");
+            abort_on_err(driver::phase_3_run_analysis_passes(&sess,
+                                                             &cstore,
+                                                             ast_map,
+                                                             &arenas,
+                                                             &id,
+                                                             resolve::MakeGlobMap::No,
+                                                             |tcx, mir_map, _, _| {
+                let mir_map = mir_map.unwrap();
+
+                for (nodeid, mir) in &mir_map.map {
+                    try!(writeln!(out, "MIR for {}", tcx.map.node_to_string(*nodeid)));
```

Comment:
Nit: I personally think the MIR pretty writer should be adapted to somehow work on MIR maps by e.g. outputting

```
fn main::main() -> ... {
...
}
```

for functions and

```
λ main::main::1(... {
}
```

for closures as well as improving the way closures are printed.